File: examples/scala/src/main/scala/com/xebia/functional/xef/scala/auto/Movie.scala
package com.xebia.functional.xef.scala.conversation

import com.xebia.functional.xef.scala.conversation.*
import io.circe.Decoder
import com.xebia.functional.xef.prompt.Prompt

private final case class Movie(title: String, genre: String, director: String) derives SerialDescriptor, Decoder

@main def runMovie: Unit =
  conversation {
    val movie = prompt[Movie](Prompt("Inception movie genre and director."))
    println(s"The movie ${movie.title} is a ${movie.genre} film directed by ${movie.director}.")
  }

File: examples/kotlin/src/main/kotlin/com/xebia/functional/xef/auto/Movie.kt
package com.xebia.functional.xef.auto

import com.xebia.functional.xef.auto.llm.openai.getOrElse
import com.xebia.functional.xef.auto.llm.openai.prompt
import kotlinx.serialization.Serializable

@Serializable
data class Movie(val title: String, val genre: String, val director: String)

suspend fun main() {
    ai {
        val movie: Movie = prompt("Please provide a movie title, genre and director for the Inception movie")
        println("The movie ${movie.title} is a ${movie.genre} film directed by ${movie.director}.")
    }.getOrElse { println(it) }
}
